From 2f017775bff6ed21cc1fb6c4fc5bbea4da4f6550 Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Wed, 2 Apr 2014 12:32:01 +0800
Subject: [PATCH 1556/1825] fix: alp: pon: AES issue when enable AES first
 before configure GEM port

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit edcd3db9dc8504240133427ffa1a660a803e617b

    In the past PON module will not enable/disable AES when configure GEM port even the AES is already configrued, now enabled it.

Signed-off-by: Victor Gu <xigu@marvell.com>

Change-Id: I6b3c98727f8f9f40207114362349c421ae47efbc
Reviewed-on: http://vgitil04.il.marvell.com:8080/6858
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../mv_drivers_lsp/mv_pon/core/gpon/gponOnuApi.c   |   14 ++++++++------
 1 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/arch/arm/plat-armada/mv_drivers_lsp/mv_pon/core/gpon/gponOnuApi.c b/arch/arm/plat-armada/mv_drivers_lsp/mv_pon/core/gpon/gponOnuApi.c
index c8e7c59e..4123e1d 100644
--- a/arch/arm/plat-armada/mv_drivers_lsp/mv_pon/core/gpon/gponOnuApi.c
+++ b/arch/arm/plat-armada/mv_drivers_lsp/mv_pon/core/gpon/gponOnuApi.c
@@ -564,9 +564,6 @@ MV_STATUS onuGponApiGemConfig(MV_U32 *gemmap, E_GponIoctlGemAction action)
 								   __FILE_DESC__, __LINE__);
 							break;
 						}
-						if (onuGponDbGemPortAesGet(portId) == MV_TRUE)
-							/*update HW as well*/
-							mvOnuGponMacAesPortIdSet(portId, MV_TRUE);
 					} else {
 						rcode  = onuGponApiGemPortIdClear(portId);
 						if (rcode != MV_OK) {
@@ -575,9 +572,6 @@ MV_STATUS onuGponApiGemConfig(MV_U32 *gemmap, E_GponIoctlGemAction action)
 								   __FILE_DESC__, __LINE__);
 							break;
 						}
-						if (onuGponDbGemPortAesGet(portId) == MV_TRUE)
-							/*remove HW configuration but keep SW encryption indication*/
-							mvOnuGponMacAesPortIdSet(portId, MV_FALSE);
 					}
 
 				}
@@ -730,6 +724,10 @@ MV_STATUS onuGponApiGemPortIdConfig(MV_U32 gemPortid)
 	}
 	onuGponDbGemPortValidSet(gemPortid, MV_TRUE);
 
+	/* update HW as well */
+	if (onuGponDbGemPortAesGet(gemPortid) == MV_TRUE)
+		mvOnuGponMacAesPortIdSet(gemPortid, MV_TRUE);
+
 #ifdef MV_GPON_STATIC_GEM_PORT
 	if (staticGemPortConfigFlag == 0)
 		if (gemPortid < 4096)
@@ -772,6 +770,10 @@ MV_STATUS onuGponApiGemPortIdClear(MV_U32 gemPortid)
 	}
 	onuGponDbGemPortValidSet(gemPortid, MV_FALSE);
 
+	/* remove HW configuration but keep SW encryption indication */
+	if (onuGponDbGemPortAesGet(gemPortid) == MV_TRUE)
+		mvOnuGponMacAesPortIdSet(gemPortid, MV_FALSE);
+
 	return rcode;
 }
 
-- 
1.7.5.4

